src/python/WMCore/JobSplitting/EventAwareLumiByWork.py

```diff
@@ -0,0 +1,276 @@
+from collections import defaultdict
+
+from WMCore.DataStructs.LumiList import LumiList
+from WMCore.JobSplitting.JobFactory import JobFactory
+from WMCore.WMBS.File import File
+
+
+class EventAwareLumiByWork(JobFactory):
+    """
+    Split jobs by lumis taking into account events per lumi
+    """
+
+    locations = []
+
+    def algorithm(self, *args, **kwargs):
+        """
+        _algorithm_
+
+        Split files into a number of lumis per job
+        Allow a flag to determine if we split files between jobs
+        """
+
+        # TODO: This has one possible weakness. If two files contain the same run/lumi section but are split across
+        # sites
+        # TODO: the job will be instructed to read a file not at the site
+
+        avgEventsPerJob = int(kwargs.get('events_per_job', 5000))
+        lumiEventLimit = int(kwargs.get('max_events_per_lumi', 20000))
+        totalEventLimit = int(kwargs.get('total_events', 0))
+        splitOnFile = bool(kwargs.get('halt_job_on_file_boundaries', True))
+        ignoreACDC = bool(kwargs.get('ignore_acdc_except', False))
+        collectionName = kwargs.get('collectionName', None)
+        splitOnRun = kwargs.get('splitOnRun', True)
+        getParents = kwargs.get('include_parents', False)
+        runWhitelist = kwargs.get('runWhitelist', [])
+        runs = kwargs.get('runs', None)
+        lumis = kwargs.get('lumis', None)
+        deterministicPileup = kwargs.get('deterministicPileup', False)
+
+        # Set the lumi mask for the fileset based on runs & lumis or runWhitelist or both
+        lumiMask = LumiList()
+        if runs and lumis and runWhitelist:
+            lumiMask = LumiList(wmagentFormat=(runs, lumis)) & LumiList(runs=runWhitelist)
+        elif runs and lumis:
+            lumiMask = LumiList(wmagentFormat=(runs, lumis))
+        elif runWhitelist:
+            lumiMask = LumiList(runs=runWhitelist)
+
+        # EWV Not looked at yet
+        self.perfParameters = kwargs.get('performance', {})
+        eventsPerLumiInDataset = 0
+
+        if deterministicPileup and self.package == 'WMCore.WMBS':
+            getJobNumber = self.daoFactory(classname="Jobs.GetNumberOfJobsPerWorkflow")
+            jobNumber = getJobNumber.execute(workflow=self.subscription.getWorkflow().id)
+            self.nJobs = jobNumber
+        # EWV End not looked at yet
+
+        # If we have runLumi info, we need to load it from couch
+        if collectionName:
+            lumiMask = self.lumiMaskFromACDC(couchURL=kwargs.get('couchURL'), couchDB=kwargs.get('couchDB'),
+                                             filesetName=kwargs.get('filesetName'), collectionName=collectionName,
+                                             owner=kwargs.get('owner'), group=kwargs.get('group'))
+
+        lDict = self.sortByLocation()
+
+        # First we need to load the data
+        if self.package == 'WMCore.WMBS':
+            loadRunLumi = self.daoFactory(classname="Files.GetBulkRunLumi")
```

Comment:
What do you mean? That the code I have doesn't do anything with it or that this is actually not needed because we don't use this mode of operations? I didn't investigate it, this is just a snippet of code left over from EventAwareByLumi.